storey/aggregations.py

```diff
@@ -611,14 +671,15 @@ def aggregate(self, timestamp, value):
     def get_features(self, timestamp):
         result = {}
 
-        args_results = [list(bucket.get_features(timestamp).values()) for bucket in self.args]
-
-        for i in range(len(args_results[0])):
-            window_string = self.window.windows[i][1]
+        args_results = [bucket.get_features(timestamp, get_hidden=True) for bucket in self.args]
+        for j in range(len(self.window.windows)):
+            window_string = self.window.windows[j][1]
             current_args = []
-            for window_result in args_results:
-                current_args.append(window_result[i])
-
+            for i in range(len(args_results)):
+                for key, value in args_results[i].items():
+                    if key.endswith('_' + window_string):
```

Comment:
by now you probably know exactly the "feature's" name so you can access it by key rather than for and if, no?